keras/callbacks.py

```diff
@@ -2717,7 +2717,7 @@ def on_train_batch_end(self, batch, logs=None):
                 step=self._train_step,
             )
 
-        if logs:
+        if logs and self.update_freq != "epoch":
```

Comment:
> [This](https://github.com/keras-team/keras/blob/master/keras/integration_test/distributed_training_test.py) and [this](https://github.com/keras-team/keras/blob/master/keras/integration_test/parameter_server_custom_training_loop_test.py) are two tests you can refer to to create a locally-run unit test that uses `ParameterServerStrategy`. Mind checking with that?

Yes, those were the tests I tried extending before attempting to enter a `ParameterServerStrategy` scope in `callbacks.py`, but I kept getting build failures when using bazel, and getting `Cannot determine binary path` RuntimeErrors when trying to run `python distributed_training_test.py` directly.

Maybe I'll take another stab at it later if I can find the time, but in the meantime, hopefully a patch like this can be more useful than not having `update_freq` at all.

Reply:
I think we would need to make sure it runs with all strategies before we can merge it to avoid breakages. In the meantime, would it work for you if you subclass the TensorBoard callback and log at batch boundaries for your use case?